plasmapy/utils/decorators/lite_func.py

```diff
@@ -0,0 +1,132 @@
+"""
+Module for defining functionality that marks and handle Lite-Function
+creation.
+"""
+__all__ = ["bind_lite_func"]
+
+import functools
+import inspect
+
+from numba.extending import is_jitted
+from typing import Callable, Dict
+from warnings import warn
+
+from plasmapy.utils.exceptions import PlasmaPyWarning
+
+
+class _LiteFuncDict(dict):
+    """
+    Dictionary of Lite-Function functionality bound to the parent
+    function.  The dictionary key is a string indicating the name
+    the functionality was bound with and the dictionary value is a
+    string containing the fully qualified path of the original
+    functionality.
+    """
+
+    # This is only to give __bound_lite_func__ a docstring.
+
+
+def bind_lite_func(lite_func, attrs: Dict[str, Callable] = None):
+    """
+    Decorator to bind a lightweight "lite" version of a formulary
+    function to the full formulary function, as well as any supporting
+    attributes.
+
+    Parameters
+    ----------
+    lite_func: Callable
+        The lightweight function to be bound as the ``lite`` attribute
+        to the function being decorated.
+
+    attrs: Dict[str, Callable]
+        A dictionary where the key is a string defining the bound name
+        and the associated value is the functionality to be bound.
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+        def foo_lite(x)
+            return x
+
+        def bar():
+            print("Supporting function.")
+
+        @bind_lite_func(foo_lite, attrs=[("bar", bar),])
+        def foo(x):
+            if not isinstance(x, float):
+                raise TypeError("Argument x can only be a float.")
+            return x
+
+        >>> foo(5)  # doctest: +SKIP
+        5
+        >>> foo.lite(5)  # doctest: +SKIP
+        5
+        >>> foo.bar()  # doctest: +SKIP
+        Supporting function.
```

Comment:
For things like this I really like using metasyntactic variables to illustrate that these should be replaced with your own code.  I use `foo` and `bar` because they're the most common and are found in most of the stack exchange examples out there, and as a result users would be familiar with them.  I would be more inclined to switch them to `spam`, `ham`, and `eggs` since that's the common Python metasyntactic variables.

[Metasyntactic variables](https://en.wikipedia.org/wiki/Metasyntactic_variable)